designs/cloudstack-multiple-endpoints.md

```diff
@@ -163,6 +177,27 @@ api-url    = http://172.16.0.3:8080/client/api
 ...
 ```
 
+1. `Global`, `AvailabilityZone2`, and `AvailabilityZone3` are entirely arbitrary names that are unnecessary.
+2. If these tags mismatch between the name in the AvailabilityZone and here there will be an error.
+3. If we use validation, then we can prevent mismatched names, but what about modifying this secret in an existing cluster?
+    - Validation doesn't operate on an existing cluster unless the secret is updated through EKS-A.
+    - If an existing cluster wants to update a secret, the entire secret must be modified together. If this results in
+    an error, the entire set of AvalailabilityZones are at risk.
+4. To do validation of the secret ini with multiple tags the DataCenter needs to match the tags to the names of its
+   AvailabilityZones. This is fine, but with yaml the secrets can be iterated on and checked individually without
+   concern for the tags.
+5. CAPC wants a k8s secret anyway.
+6. Currently we create a k8s secret in a highly convoluted maner that requires b64 encoding an environment
+   variable that is expanded during clusterctl expansion but decoded in the CAPC controllers. The b64
+   encoding provides no extra security. It is used purely as a way to avoid potential command line special characters 
+   present in the cloud-config ini file.
+7. Other providers _will_ need multiple secrets soon enough. One in the pipeline already requires a CSV as a format to
+    provide details. We should standardize on a general way to pass multiple secrets per provider. The data present in
+    these secrets should be entirely provider specific. In this manner any arbitrary data form that a specific provider
```

Comment:
Could you elaborate on what other providers you're referring to here? Additionally, which pipeline are you referring to?

Reply:
I'll check with Vignesh. Pipeline meaning other providers that are not yet GA.

One of them requires a CSV per Node.